src/insert/spacefinder/article.ts

```diff
@@ -492,34 +495,26 @@ const attemptToAddInlineMerchAd = (): Promise<boolean> => {
 	});
 };
 
-const doInit = async (): Promise<boolean> => {
+/**
+ * Init all the article body adverts, including `im` and `carrot`
+ * @param fillAdSlot a function to fill the ad slots
+ */
+const init = async (fillAdSlot: FillAdSlot): Promise<boolean> => {
 	if (!commercialFeatures.articleBodyAdverts) {
 		return Promise.resolve(false);
 	}
 
 	const im = window.guardian.config.page.hasInlineMerchandise
-		? attemptToAddInlineMerchAd()
+		? attemptToAddInlineMerchAd(fillAdSlot)
 		: Promise.resolve(false);
 	const inlineMerchAdded = await im;
 	if (inlineMerchAdded) await waitForAdvert('dfp-ad--im');
 
-	await addInlineAds();
+	await addInlineAds(fillAdSlot);
 
 	await initCarrot();
 
 	return im;
 };
 
-/**
- * Initialise article body ad slots
- */
-export const init = (): Promise<boolean> => {
-	// Also init when the main article is redisplayed
-	// For instance by the signin gate.
-	mediator.on('page:article:redisplayed', doInit);
```

Comment:
Mediator hasn't worked for a significant amount of time, I think we should remove this.